src/devices/cpu/sh/sh7604.h

```diff
@@ -0,0 +1,283 @@
+	{
+		COMBINE_DATA(&m_vcrdma[Channel]);
+		m_irq_vector.dmac[Channel] = m_vcrdma[Channel] & 0x7f;
+		sh2_recalc_irq();
+	}
+
+	template <int Channel> uint8_t drcr_r() { return m_dmac[Channel].drcr & 3; }
+	template <int Channel> void drcr_w(uint8_t data) { m_dmac[Channel].drcr = data & 3; sh2_recalc_irq(); }
+	template <int Channel> uint32_t sar_r() { return m_dmac[Channel].sar; }
+	template <int Channel> void sar_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0) { COMBINE_DATA(&m_dmac[Channel].sar); }
+	template <int Channel> uint32_t dar_r() { return m_dmac[Channel].dar; }
+	template <int Channel> void dar_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0) { COMBINE_DATA(&m_dmac[Channel].dar); }
+	template <int Channel> uint32_t dmac_tcr_r() { return m_dmac[Channel].tcr; }
+	template <int Channel> void dmac_tcr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0) { COMBINE_DATA(&m_dmac[Channel].tcr); m_dmac[Channel].tcr &= 0xffffff; }
+	template <int Channel> uint32_t chcr_r() { return m_dmac[Channel].chcr; }
+	template <int Channel> void chcr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0)
+	{
+		uint32_t old;
+		old = m_dmac[Channel].chcr;
+		COMBINE_DATA(&m_dmac[Channel].chcr);
+		m_dmac[Channel].chcr = (data & ~2) | (old & m_dmac[Channel].chcr & 2);
+		sh2_dmac_check(Channel);
+	}
+	uint32_t dmaor_r() { return m_dmaor & 0xf; }
+	void dmaor_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0)
+	{
+		if(ACCESSING_BITS_0_7)
+		{
+			uint8_t old;
+			old = m_dmaor & 0xf;
+			m_dmaor = (data & ~6) | (old & m_dmaor & 6); // TODO: should this be old & data & 6? bug?
+			sh2_dmac_check(0);
+			sh2_dmac_check(1);
+		}
+	}
+
+	// WTC
+	uint16_t wtcnt_r();
+	void wtcnt_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint16_t rstcsr_r();
+	void rstcsr_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+
+	// misc
+	uint16_t fmr_sbycr_r();
+	void fmr_sbycr_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint8_t ccr_r();
+	void ccr_w(uint8_t data);
+
+	// BSC
+	uint32_t bcr1_r();
+	void bcr1_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t bcr2_r();
+	void bcr2_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t wcr_r();
+	void wcr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t mcr_r();
+	void mcr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t rtcsr_r();
+	void rtcsr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t rtcor_r();
+	void rtcor_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t rtcnt_r();
+	void rtcnt_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+
+	virtual void set_frt_input(int state) override;
+
+	// SCI
+	uint8_t m_smr = 0, m_brr = 0, m_scr = 0, m_tdr = 0, m_ssr = 0;
+	// FRT / FRC
+	uint8_t m_tier = 0, m_ftcsr = 0, m_frc_tcr = 0, m_tocr = 0;
+	uint16_t m_frc = 0;
+	uint16_t m_ocra = 0, m_ocrb = 0, m_frc_icr = 0;
```

Comment:
Can you please move initialisation for data members that aren’t part of mini nested structures/classes to the constructor initialiser list?  Having them in the header increases compilation overhead and fragility.

Reply:
Yeah, sounds good to me. I debated with myself on whether I should make that change in this PR since I vaguely remember you commenting something similar in the past on someone else's commit.